component/prometheus/operator/podmonitors/config_gen/config_gen_podmonitor.go

```diff
@@ -0,0 +1,262 @@
+package config_gen
+
+// SEE https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/promcfg.go
+
+import (
+	"fmt"
+	"net/url"
+	"sort"
+	"strings"
+
+	v1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	namespacelabeler "github.com/prometheus-operator/prometheus-operator/pkg/namespace-labeler"
+	commonConfig "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+	promk8s "github.com/prometheus/prometheus/discovery/kubernetes"
+	"github.com/prometheus/prometheus/model/relabel"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	regexFilterRunning = relabel.MustNewRegexp("(Failed|Succeeded)")
+	regexTrue          = relabel.MustNewRegexp("true")
+	regexAnything      = relabel.MustNewRegexp("(.+)")
+)
+
+func (cg *ConfigGenerator) GeneratePodMonitorConfig(m *v1.PodMonitor, ep v1.PodMetricsEndpoint, i int) (cfg *config.ScrapeConfig, err error) {
```

Comment:
I think the implementation of this function can probably be refactored a little bit to be smaller, but I'm not going to block the PR on this since the PR is big enough.  

Maybe the next PR for another component can take the opportunity to refactor?

Reply:
sounds good, we will refactor this once we add service monitor support